documentation/architecture/README.md

```diff
@@ -0,0 +1,263 @@
+# Beta.Grants.Gov Architecture
+
+This document is meant to be a living record of the architecture for the beta.grants.gov system. This includes the application, network, and infrastructure architecture, as well as the CI/CD pipeline, and other services and integrations used to support the applications.
+
+At a high level, this system uses Github to maintain the codebase repository and run the CI/CD pipeline, and AWS to host the applications and its supporting services.
+
+## Architecture
+This is a general architecture diagram of the beta.grants.gov system.
+
+```mermaid
+%%{init: {'theme': 'neutral' } }%%
+flowchart TB
+
+    %% AWS Tenant
+    subgraph AWS [HHS AWS Tenant]
+        shared
+        VPC:::az
+    end
+
+    %% AWS Shared Services
+    subgraph shared [AWS Shared Services]
+        ECSC
+        IAM:::sec
+        SSM:::sec
+        CloudWatch:::sec
+        ECR:::ecs
+    end
+
+    subgraph ECSC [ECS Cluster]
+        ECSS["ECS Service
+        Fargate Launch Type"]:::ecs
+    end
+    style ECSC stroke:#FF9900
+
+    %% AWS Services Within VPC
+    subgraph VPC ["AWS Virtual Private Cloud (VPC)"]
+        direction LR
+        AZ1:::az
+        AZ2:::az
+        public-subnet1:::subnet
+        RDS
+        AZ1 & AZ2 --> RDS
+    end
+
+    subgraph public-subnet1 [Public Subnet]
+        ALB["Application Load Balancer (ALB)"]:::lb --> AZ1 & AZ2
+    end
+
+    subgraph AZ1 [Availability zone 1]
+        direction TB
+        private-subnet1:::subnet
+        subgraph private-subnet1 [Private Subnet]
+            ECS1:::ecs
+            ECS2:::ecs
+        end
+
+    end
+    subgraph AZ2 [Availability zone 2]
+        direction TB
+        private-subnet2:::subnet
+        subgraph private-subnet2 [Private Subnet]
+            ECS3:::ecs
+            ECS4:::ecs
+        end
+
+    end
+    subgraph ECS1 ["ECS Front-End Task (Next.js)"]
+        c1[" Docker
+        Container 1"]
+        c2[" Docker
+        Container n"]
+    end
+    subgraph ECS2 ["ECS Back-End Task (Flask)"]
+        c3[" Docker
+        Container 1"]
+        c4[" Docker
+        Container n"]
+    end
+    subgraph ECS3["ECS Front-End Task (Next.js)"]
+        c5[" Docker
+        Container 1"]
+        c6[" Docker
+        Container n"]
+    end
+    subgraph ECS4 ["ECS Back-End Task (Flask)"]
+        c7[" Docker
+        Container 1"]
+        c8[" Docker
+        Container n"]
+    end
+
+    subgraph RDS
+        subgraph private-subnet3 [Private Subnet]
+            DB[("Multi-AZ Grants.gov
+            Replica Database
+            tstgrntsee1.cshqswcusbcd.
+            us-east-1.rds.amazonaws.com
+            :1521")]:::db
```

Comment:
I know this is in a private subnet so there shouldn't necessarily be a security concern, but my instinct would be avoid hard coding IP addresses and ports into the diagram -- unless you feel like including those details here provides important system-design-level information.

Reply:
I would agree w/ removing the actual address.